### PR TITLE
Release 2.20

### PR DESCRIPTION
**Pull request by Ralim**
Preparing for 2.20 release

### CODE REVIEW

**Comment by discip** [comment]
@Ralim
I think we should update all the translation files prior to releasing this, because the actual description text for the CJC calibration confirmation might be a bit misleading.

The current `SettingsCalibrationWarning` is:
*`"Please make sure tip & handle are at room temperature`**`before calibration`**`!"`*

... but should be:
*`"Please make sure tip & handle are at room temperature`**`at next boot`**`!"`*

I have tried doing this in my fork but unfortunately this increases the size to no longer fit the **`TS80P`**. 😭

Do you have any idea?

**Comment by Ralim** [comment]
@discip 

I'm happy for translations to be updated. I just pushed some changes to buy a few bytes in TS80P; see if your changes work now :)

**Comment by discip** [comment]
@Ralim
Will definitely do as soon as I am home. 😅

**Comment by discip** [comment]
@Ralim
Unfortunately there is still not enough free space. 😭

**Comment by Ralim** [comment]
@discip Made a tiny bit more space, does that fit yet?

**Comment by Ralim** [comment]
@discip Okay no rush but you should have room now :)

**Comment by discip** [comment]
@Ralim
Now it fits. 😃 :thumbsup:

I am on it right now.
Need a bit time to include all translators & for the documentation.

**Comment by Ralim** [comment]
No rush, let me know how it goes.

**Comment by Ralim** [comment]
Alrighty this is overdue; anyone against this going out ?

**Comment by discip** [comment]
I would like to change this:
https://github.com/Ralim/IronOS/blob/628b8fe32d81ca055683506c016f7504eaa1c6af/Translations/translation_EN.json#L300
back to:
```json
"desc": "Calibrates tip Cold Junction Compensation at the next boot (not required if Delta T is < 5°C)"
```
since it did not bring the anticipated success and is somewhat *unclear*.

But I'm afraid to annoy the translators with the frequency of changes.

**Comment by discip** [comment]
Before the merge,
I would like to change this:
https://github.com/Ralim/IronOS/blob/628b8fe32d81ca055683506c016f7504eaa1c6af/Translations/translation_EN.json#L300
back to:
```json
"desc": "Calibrates tip Cold Junction Compensation at the next boot (not required if Delta T is < 5°C)"
```
since it did not bring the anticipated success and is somewhat *unclear*.

But I'm afraid to annoy the translators with the frequency of changes.

**Comment by discip** [comment]
Before the merge,
I would like to change this:
https://github.com/Ralim/IronOS/blob/628b8fe32d81ca055683506c016f7504eaa1c6af/Translations/translation_EN.json#L300
back to:
```json
"desc": "Calibrates tip Cold Junction Compensation at the next boot (not required if Delta T is < 5°C)"
```
since it did not bring the anticipated success and is somewhat *unclear*.

But I'm afraid to annoy the translators with the frequency of changes.
(Currently only federicodilo & dmitrygribenchuk are affected.)

**Comment by discip** [comment]
Before the merge,
I would like to change this:
https://github.com/Ralim/IronOS/blob/628b8fe32d81ca055683506c016f7504eaa1c6af/Translations/translation_EN.json#L300
back to:
```json
"desc": "Calibrates tip Cold Junction Compensation at the next boot (not required if Delta T is < 5°C)"
```
since it did not bring the anticipated success and is somewhat *unclear*.

But I'm afraid to annoy the translators with the frequency of changes.
(Currently only federicodilo & dmitrygribenchuk are affected.)

In general, I would like to revise and unify the translation one more time. 😅
Or should I wait with that?

**Comment by Ralim** [comment]
Go for it by me

File: Translations/translation_NL.json [hunk 1]
```diff
@@ -1,339 +1,345 @@
...
+{
+	"languageCode": "NL",
+	"languageLocalName": "Nederlands",
+	"fonts": [
+		"ascii_basic",
+		"latin_extended"
+	],
+	"tempUnitFahrenheit": false,
+	"messages": {
+		"SettingsCalibrationWarning": "Before rebooting, make sure tip & handle are at room temperature!",
```
Comment (KoopsF) [inline Translations/translation_NL.json:10]: Suggestion: "Zorg ervoor dat voor de volgende keer opstarten de punt én het handvat op kamertemperatuur zijn!"

**Review by KoopsF — CHANGES_REQUESTED** [review]
translation suggestion

**Comment by discip** [comment]
Will do, as soon as I am home again. 😊

**Comment by Firebie** [comment]
```
diff --git a/source/Core/BSP/Miniware/ThermoModel.cpp b/source/Core/BSP/Miniware/ThermoModel.cpp
index 1b04bd5f..a7bc8094 100644
--- a/source/Core/BSP/Miniware/ThermoModel.cpp
+++ b/source/Core/BSP/Miniware/ThermoModel.cpp
@@ -125,6 +125,6 @@ const int32_t uVtoDegC[] = {
     38137, 500, //
 };
 #endif
-const int uVtoDegCItems = sizeof(uVtoDegC) / (2 * sizeof(uint16_t));
+const int uVtoDegCItems = sizeof(uVtoDegC) / (2 * sizeof(int32_t));
 
 uint32_t TipThermoModel::convertuVToDegC(uint32_t tipuVDelta) { return Utils::InterpolateLookupTable(uVtoDegC, uVtoDegCItems, tipuVDelta); }
diff --git a/source/Core/BSP/Pinecil/ThermoModel.cpp b/source/Core/BSP/Pinecil/ThermoModel.cpp
index 846a0a00..4515c42f 100644
--- a/source/Core/BSP/Pinecil/ThermoModel.cpp
+++ b/source/Core/BSP/Pinecil/ThermoModel.cpp
@@ -67,6 +67,6 @@ const int32_t uVtoDegC[] = {
 };
 #endif
 
-const int uVtoDegCItems = sizeof(uVtoDegC) / (2 * sizeof(uint16_t));
+const int uVtoDegCItems = sizeof(uVtoDegC) / (2 * sizeof(int32_t));
 
 uint32_t TipThermoModel::convertuVToDegC(uint32_t tipuVDelta) { return Utils::InterpolateLookupTable(uVtoDegC, uVtoDegCItems, tipuVDelta); }
```

**Comment by alvinhochun** [comment]
> ```
> diff --git a/source/Core/BSP/Miniware/ThermoModel.cpp b/source/Core/BSP/Miniware/ThermoModel.cpp
> index 1b04bd5f..a7bc8094 100644
> --- a/source/Core/BSP/Miniware/ThermoModel.cpp
> +++ b/source/Core/BSP/Miniware/ThermoModel.cpp
> @@ -125,6 +125,6 @@ const int32_t uVtoDegC[] = {
>      38137, 500, //
>  };
>  #endif
> -const int uVtoDegCItems = sizeof(uVtoDegC) / (2 * sizeof(uint16_t));
> +const int uVtoDegCItems = sizeof(uVtoDegC) / (2 * sizeof(int32_t));
>  
>  uint32_t TipThermoModel::convertuVToDegC(uint32_t tipuVDelta) { return Utils::InterpolateLookupTable(uVtoDegC, uVtoDegCItems, tipuVDelta); }
> diff --git a/source/Core/BSP/Pinecil/ThermoModel.cpp b/source/Core/BSP/Pinecil/ThermoModel.cpp
> index 846a0a00..4515c42f 100644
> --- a/source/Core/BSP/Pinecil/ThermoModel.cpp
> +++ b/source/Core/BSP/Pinecil/ThermoModel.cpp
> @@ -67,6 +67,6 @@ const int32_t uVtoDegC[] = {
>  };
>  #endif
>  
> -const int uVtoDegCItems = sizeof(uVtoDegC) / (2 * sizeof(uint16_t));
> +const int uVtoDegCItems = sizeof(uVtoDegC) / (2 * sizeof(int32_t));
>  
>  uint32_t TipThermoModel::convertuVToDegC(uint32_t tipuVDelta) { return Utils::InterpolateLookupTable(uVtoDegC, uVtoDegCItems, tipuVDelta); }
> ```

I would say you should use `sizeof(uVtoDegC) / (2 * sizeof(uVtoDegC[0]))`. But to be frank, I would probably declare `uVtoDegC` to be `struct { int32_t a, int32_t b }` so you can write `sizeof(uVtoDegC) / sizeof(uVtoDegC[0])`.

**Comment by Ralim** [comment]
@discip Should I just release this as is?

**Comment by discip** [comment]
@Ralim
Apart from the fact that I would like to see #1494 implemented, there is nothing hindering that from happening.

**Pull request merged**